database/db.js

```diff
@@ -19,7 +19,6 @@ module.exports = {
         id_number: `${studentObj.last_name}.${ await module.exports.getLastNameCount(studentObj.last_name)}`,
         program_list: studentObj.program_list,
```

Comment:
When saving the Student object here, you'll also want to update each of the Program objects and add this student to their `student_list`.

**Side-note**
Alternatively, you could change the design so that there is a "Linked Table" type structure, so instead of storing a list of programs on each student and a list of students on each program, you'd just store a "Registration" for each of these links. Each registration would simply have a student ID and a program ID, and you could use those objects to lookup relationships. [This article](https://fmhelp.filemaker.com/help/18/fmp/en/index.html#page/FMP_Help/many-to-many-relationships.html) has some more information about this from a relational database perspective.

Reply:
You'll also want to update the `editStudentById` function to do the same thing - update any programs that have changed. If a student is unenrolled in a Program, they should be removed from that Program's `student_list`. If they are newly enrolled in a Program, they should be added to that Program's `student_list`.